multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -64,7 +93,26 @@ func (c *Cluster) stackName() string {
 	return c.cfg.ClusterName
 }
 
-func (c *Cluster) Create(tlsConfig *TLSConfig) error {
+func (c *Cluster) initAssets(assetDir string) *TLSConfig {
+
+	c.cfg.InstallWorkerScript = blobutil.MustReadCompressFile(filepath.Join(assetDir, "scripts", "install-worker.sh"))
+	c.cfg.InstallControllerScript = blobutil.MustReadCompressFile(filepath.Join(assetDir, "scripts", "install-controller.sh"))
+
+	manifestPath := filepath.Join(assetDir, "manifests")
+	c.cfg.ClusterManifestsTar = blobutil.MustTarCompressDirectory(assetDir, filepath.Join(manifestPath, "cluster"))
+	c.cfg.ControllerManifestsTar = blobutil.MustTarCompressDirectory(assetDir, filepath.Join(manifestPath, "controller"))
+	c.cfg.WorkerManifestsTar = blobutil.MustTarCompressDirectory(assetDir, filepath.Join(manifestPath, "worker"))
+
+	credentialsDir := filepath.Join(assetDir, "credentials")
+	tlsConfig := NewTLSConfig(credentialsDir)
+	tlsConfig.ReadFilesFromPaths()
+	return tlsConfig
+}
+
+func (c *Cluster) Create(assetDir string) error {
+
+	tlsConfig := c.initAssets(assetDir)
```

Comment:
Agreed on that. We have a lot of complexity dangling outside the black box for no good reason.